server/dataone_register.py

```diff
@@ -242,3 +244,41 @@ def D1_lookup(path):
         'repository': 'DataONE',
     }
     return dataMap
+
+
+def get_file_list(package_pid):
+    """
+    Retrieve a list of all the files in a data package.The metadata
+    record providing information about the package is also in this list.
+    """
+
+    result = query('resourceMap:"{}"'.format(esc(package_pid)),
+                   ["identifier", "formatType", "title", "size", "formatId",
+                    "fileName", "documents"])
+
+    if 'response' not in result or 'docs' not in result['response']:
+        raise RestException(
+            "Failed to get a result for the query\n {}".format(result))
+
+    return result['response']['docs']
+
+
+def get_package_list(path):
+    """
+    Retrieves a list of all the files in a package with
+    their sizes
+    """
+
+    package_pid = get_package_pid(path)
+    logger.debug('Found package PID of {}.'.format(package_pid))
+
+    docs = get_file_list(package_pid)
+    files = list()
+    for doc in docs:
+        data_map = {
+            'name': doc.get('title', doc.get('identifier')),
```

Comment:
I think this should be `doc.get('fileName', doc.get('identifier')`. `title` is only for metadata Objects anyway.